tests/test_servefile.py

```diff
@@ -117,6 +118,22 @@ def check_download(expected_data=None, path='/', fname=None, status_code=200, **
     return r  # for additional tests
 
 
+def _retry_while(exception, function, timeout=2):
+    now = datetime.datetime.now
```

Comment:
I know this might be kind of academic, but I think using `datetime.datetime.now()` in this function might let the tests fail during DST passover. `now()` gives us a tz-less "naive" timestamp. If `timeout_after` is set before a timeswitch now() is executed after a +1h timeswitch the method would timeout immediate and the test would fail.

How about using `time.time()` here? That would be a tz-less monotonic alternative that would just gives us the elapsed seconds.

Reply:
I don't mind taking a monotonic clock into consideration, but with Python 2.7 support not removed we cannot use `time.monotonic` and `time.time` does not seem to have guarantees like that, that I could find. Options I see are:

- a) Research and proof that `time.time`is monotonic, then use that
- b) Drop Python 2.7 support in another pull request, merge that, use `time.monotonic`, rebase onto latest `master`
- c) Be okay with the DST issue

How would you like to continue?

Reply:
`datetime.datetime.now()` returns the current time of the computer, stripping all timezone info. For example 
when the computer switches from CET to CEST for python time time will have jumped +1h. `time.time()` on the other hand is always UTC and therefore not subject to local timezone changes. So conceptually I'd go with `time.time()`. This behaviour would also what I'd be expect to be the case with python2.

I haven't read how this works in python, but monotonic clocks are as far as I know a concept of time measurement where it's promised (guaranteed?) that the clock will never wind back. This might be important in the (rarer) case that something in the system resets the system clock to another value without letting it "slide" to the new time.

I'd go with the `time.time()` version, that'd be good enough for me.

Reply:
I personally don't see the point of fixing part of the monotonicity problem without fixing the whole of it, but I have made the code use `time.time` for you just now.

Reply:
Thanks! I feel much more comfortable with `time.time()`. If you're still interested do discuss the intricacies of this after the PR I'd also be open for that. :)